src/components/new-safe/create/logic/index.ts

```diff
@@ -207,15 +213,20 @@ export const showSafeCreationError = (error: EthersError | Error): AppThunk => {
 }
 
 export const checkSafeCreationTx = async (
-  provider: JsonRpcProvider,
+  provider: Provider,
   pendingTx: PendingSafeTx,
   txHash: string,
   dispatch: AppDispatch,
 ): Promise<SafeCreationStatus> => {
   const TIMEOUT_TIME = 60 * 1000 // 1 minute
 
   try {
-    const receipt = await provider._waitForTransaction(txHash, 1, TIMEOUT_TIME, pendingTx)
+    const receipt = await provider.waitForTransaction(txHash, 1, TIMEOUT_TIME)
+
+    /** The receipt should always be non-null as we require 1 confirmation */
+    if (receipt === null) {
+      throw SafeCreationStatus.ERROR
```

Comment:
Well, my reasoning was:
1. null can't happen if you look at the waitForTransaction as we pass `1` cofirms
2. if it somehow happens, then all bets are off and I have no idea what has happened here, so best thing in my mind would to blow(sorry, throw) this thing off :)

Reply:
Gotcha. If it's not cauught, it will crash the app, so we'll see if it ever happens on Sentry. Just make sure to wrap it in an Error, otherwise it will throw a string.